Data migration console app updates

s3364859 opened the pull request:
@6nop @s3494110 @s3482247 

In this PR:
 - Package updates
 - Remove EF stuff
 - Fix up migrator console app
 - Add in test migration as an example
 - Move some common code into a new project
 - Retarget .NET 4.5.2 ->4.6.1

![move-along](https://cloud.githubusercontent.com/assets/22001846/18381780/38dfcfa2-76c1-11e6-8c4b-7c3fcaac8543.gif)

s3364859 commented on CPT331.Core/Logging/OutputStreams.cs:
```diff
@@ -0,0 +1,31 @@
+
+namespace CPT331.Core.Logging
+{
+	public static class OutputStreams
```
New type for logging information to various output streams. Might be able to include some logging to the file system here too.

6nop replied:
I'm happy with this.

s3364859 commented on CPT331.Data/Test.cs:
```diff
@@ -0,0 +1,46 @@
+{
+	public class Test
+	{
+		public Test(int id, string value0, string value1)
```
This type models a row in dbo.Table.

s3364859 commented on CPT331.Data/TestRepository.cs:
```diff
@@ -0,0 +1,31 @@
+
+			using (SqlConnection sqlConnection = SqlConnectionFactory.NewSqlConnetion())
+			{
+				getTests = SqlMapper
```
Here's how you use Dapper to hydrate objects.

6nop replied:
Seems simple enough. SPs look like a good way to avoid to much querying in code (which is difficult to test/maintain).

s3364859 commented:
@6nop Thanks for looking at that mate. I've updated the project files to remove that binding, and have also removed and EF related stuff from the config files and replaced it with the connection string.

Also, I agree with you, we need a better way of dealing with identity when it comes to the connection string. I wonder if going with Windows auth will be good enough?

6nop commented:
@s3364859 Thanks for making the changes.

The projects now build and launch without errors/warnings.

WinAuth would be ideal. On the other hand, perhaps the current sa user (assuming we all use the same password) will be easier for those less familiar with IIS / Sql Server. 

So I guess I'm happy to leave 'sa' as is for now. Maybe we can just do a config transformation for the published site.
